interceptor.go

```diff
@@ -50,47 +52,86 @@ func (i *Interceptor) WrapUnary(next connect.UnaryFunc) connect.UnaryFunc {
+			}
+			reporter.ReportStarted(callType, callPackage, callMethod)
+			defer func() { reporter.ReportHandled(callType, callPackage, callMethod, code) }()
+			defer func() {
+				reporter.ReportHandledSeconds(callType, callPackage, callMethod, code, time.Since(now).Seconds())
+			}()
 		}
 
 		resp, err := next(ctx, req)
-		code := codeOf(err)
-		var respMsg any
+		code = codeOf(err)
 		if err == nil {
-			respMsg = resp.Any()
-		}
-
-		if reporter != nil {
-			reporter.ReportHandled(callType, callPackage, callMethod, code, respMsg)
-			reporter.ReportHandledSeconds(callType, callPackage, callMethod, code, time.Since(now).Seconds())
+			var bytes *prom.CounterVec
+			if reporter.isClient {
+				bytes = reporter.bytesSent
+			} else {
+				bytes = reporter.bytesReceived
+			}
+			if bytes != nil {
+				bytes.WithLabelValues(callType, callPackage, callMethod).Add(float64(proto.Size(resp.Any().(proto.Message))))
+			}
 		}
 
 		return resp, err
 	})
 }
 
 func (i *Interceptor) WrapStreamingClient(next connect.StreamingClientFunc) connect.StreamingClientFunc {
-	return connect.StreamingClientFunc(func(ctx context.Context, s connect.Spec) connect.StreamingClientConn {
-		conn := next(ctx, s)
-		if i.client != nil {
-			conn = newStreamingClientConn(conn, i)
+	return connect.StreamingClientFunc(func(ctx context.Context, spec connect.Spec) connect.StreamingClientConn {
+		// Short-circuit, not configured to report for client.
+		if i.client == nil {
+			return next(ctx, spec)
+		}
+
+		now := time.Now()
+		callType := streamTypeString(spec.StreamType)
+		callPackage, callMethod := procedureToPackageAndMethod(spec.Procedure)
+
+		i.client.ReportStarted(callType, callPackage, callMethod)
+		onClose := func(err error) {
+			code := codeOf(err)
+			i.client.ReportHandled(callType, callPackage, callMethod, code)
+			i.client.ReportHandledSeconds(callType, callPackage, callMethod, code, time.Since(now).Seconds())
 		}
-		return conn
+
+		conn := next(ctx, spec)
+		return newStreamingClientConn(conn, i, onClose)
 	})
 }
 
 func (i *Interceptor) WrapStreamingHandler(next connect.StreamingHandlerFunc) connect.StreamingHandlerFunc {
 	return connect.StreamingHandlerFunc(func(ctx context.Context, shc connect.StreamingHandlerConn) error {
-		var newShc *streamingHandlerConn
-		if i.server != nil {
-			newShc = newStreamingHandlerConn(shc, i)
-			shc = newShc
+		// Short-circuit, not configured to report for server.
+		if i.server == nil {
+			return next(ctx, shc)
 		}
+
+		now := time.Now()
+		callType := streamTypeString(shc.Spec().StreamType)
+		callPackage, callMethod := procedureToPackageAndMethod(shc.Spec().Procedure)
+
+		var code string
+		i.server.ReportStarted(callType, callPackage, callMethod)
+		defer func() { i.server.ReportHandled(callType, callPackage, callMethod, code) }()
+		defer func() {
+			i.server.ReportHandledSeconds(callType, callPackage, callMethod, code, time.Since(now).Seconds())
+		}()
+
```

Comment:
same here, one defer

```suggestion
		callPackage, callMethod := procedureToPackageAndMethod(shc.Spec().Procedure)

		var code string
		i.server.ReportStarted(callType, callPackage, callMethod)
		defer func() {
			i.server.ReportHandled(callType, callPackage, callMethod, code)
			i.server.ReportHandledSeconds(callType, callPackage, callMethod, code, time.Since(now).Seconds())
		}()
```

<sub>_[Suggestion by Codeium](https://codeium.com/forge)_</sub>